### DISCUSSION TITLE
如何配置永久链接（在双链笔记中做源链接回溯）

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#3188] [服务]
# 重要提示

> ### 嗨，简悦全新的 [教程中心](https://www.yuque.com/kenshin/simpread) 已上线，**下面的内容并不一定是最新的**，建议 [访问这里](https://www.yuque.com/kenshin/simpread/egvhm7) 查看最新版教程。

***

> 如果你是双链笔记使用者，通过此功能可以实现：每个标注都可以追述到原文，且是在本地永久保存的原文对应的位置。

## 一个例子

> 这是 [简悦社区](https://t.me/simpreadgroup) 用户 @GreenHatHG 使用此功能制作的一个 [Logseq 模板](https://github.com/Kenshin/simpread/discussions/2153#discussioncomment-1792429) 使用流程。

https://user-images.githubusercontent.com/81074/145699931-21398cb5-c3d2-4a40-b802-8f8f6c5ebfcc.mp4

## 配置前提

- [如何让稍后读支持读取本地缓存](https://github.com/Kenshin/simpread/discussions/2146)
- [当产生标注时，如何自动同步到本地](https://github.com/Kenshin/simpread/discussions/2220)

## 配置永久链接

> 在双链笔记中的内容，可以通过链接来直接查看原文中的出处。

> 位置在选项页 → 服务 → 定制导出

![image](https://user-images.githubusercontent.com/81074/145700260-c4c8bf82-95ed-4668-a8ad-366129df82c6.png)

> 如果没有看到上图，需要开启下面的选项

![](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/simpered/kb/SCR-20220223-g93.png)

### 内部链接

> 需要改成 `http://localhost:7026/reading/` （且目前仅为此格式）。

> 此功能需要配合[ Markdown 定制化](http://ksria.com/simpread/docs/#/%E5%AE%9A%E5%88%B6%E5%8C%96%E5%AF%BC%E5%87%BA?id=markdown) 来实现，效果更好，这里是一些简悦社区用户提供的 [现成模板](https://github.com/Kenshin/simpread/discussions/2153)。

https://user-images.githubusercontent.com/81074/145700385-bb58cf64-fd10-4a6e-8a60-de846b1a3257.mp4

### 外部链接

> 功能与内部链接一致，只是可用于外部访问而已，当配置完外部链接后，即可成为 [资料库](https://github.com/Kenshin/simpread/discussions/2956)，可用于 [阅读列表](https://github.com/Kenshin/simpread/discussions/2954) 使用。

> 此功能需要开通 [开放平台](https://github.com/Kenshin/simpread/discussions/2757)，外部链接需要改为  `https://simpread.pro/@id/reading/`  `id` 需要登录开发平台里面自行设置。

https://user-images.githubusercontent.com/81074/145700765-12118b78-6adf-4a55-9db4-a7c595789e6c.mp4

## 外部链接自建方案

简悦现在可以支持外部链接自建方案，详细 [请看这里](https://github.com/Kenshin/simpread/discussions/3823)。

**Comment by itcats** [comment]
<img width="344" alt="image" src="https://user-images.githubusercontent.com/12854561/153745210-c969ec3e-527e-4dab-9893-097d0e4f6520.png">

确认按照教程一步一步操作，但是显示404，求帮助

**Reply by Kenshin** [reply]
给出的信息太少了，我没法判断。

**Reply by itcats** [reply]
按照 [logseq 搭配简悦建立自己的知识库](https://greenhathg.github.io/2021/12/12/logseq%E6%90%AD%E9%85%8D%E7%AE%80%E6%82%A6%E5%BB%BA%E7%AB%8B%E8%87%AA%E5%B7%B1%E7%9A%84%E7%9F%A5%E8%AF%86%E5%BA%93/) 的步骤进行操作，并使用了 https://github.com/Kenshin/simpread/discussions/2153#discussioncomment-1792429  Markdown 模板库中该文作者的模板，现在能实现的是下载到本地，logseq也能同步，就是点击📌打开7026本地链接时显示404，自查没发现有哪里不同，助手版本1.0.1，简悦版本2.2.0

**Reply by Kenshin** [reply]
1. 是第一个稍后读吗？如果是的话，看下这个问题 https://github.com/Kenshin/simpread/discussions/3098

2. 你的 `output` 文件夹存在本地文件吗？

**Reply by itcats** [reply]
根据#3098 解决，感谢解答。

**Reply by Kenshin** [reply]
哈哈，原来是这个问题，搞定就好。

**Reply by Soywei00** [reply]
在服务-定制化导出页面没有找到内部链接、外部链接的选项…… extension://clgdhlhfiocphghdkdbgdlmfaafccfmc/options/options.html#connect

**Comment by Leftzhf** [comment]
曾经配置好了内部链接，也没有问题。许久没用，然后今天把那个修复标题特殊字符的插件去掉了，之后标注稍后读导出到本地后发现笔记内的内部链接打开显示404，然后发现之前做的笔记内的内部链接打开都是404了，求助
![image](https://user-images.githubusercontent.com/59813926/154797207-b2980bc6-2f2e-4990-90f4-12508d33959b.png)

**Reply by Kenshin** [reply]
查看 http://localhost:7026/reading/index 看看内容？
如果有内容的话，应该是你在 `output` 里面的列表

> 之后标注稍后读导出到本地后发现笔记内的内部链接打开显示404

这个链接应该是 `http://localhost:7026/reading/xxxx` 吧？看看你在 output 里面是否存在 `xxxx-` 开头的 HTML 文件。

**Reply by Leftzhf** [reply]
![image](https://user-images.githubusercontent.com/59813926/154798267-8eb4f5f9-4ae8-4046-8dfb-294f13be8b3a.png)
这里面啥都没有，然后同步助手导出的文件夹内是有对应的html文件的

**Reply by Kenshin** [reply]
> 然后同步助手导出的文件夹内是有对应的html文件的

看看是不是同步助手导出文件夹中是不是多了一些隐藏文件？另外，你什么时候安装的同步助手？

**Reply by Leftzhf** [reply]
没见有隐藏文件，同步助手是去年12月左右装的

**Reply by Leftzhf** [reply]
不知道是不是同步的问题？刚刚打开http://localhost:7026/reading/index  的时候发现里面又有内容了,又恢复正常了。感谢解答

**Reply by Kenshin** [reply]
没问题就好。

**Reply by Kenshin** [reply]
补充下：

同步助手 1.0.1 版有个 Bug，会导致导出文件夹有隐藏文件时出现查看 http://localhost:7026/reading/index 为空的情况，**但这个错误的版本只存在了几个小时**，并且也是去年五月份发布后的几个小时内而已。

**Comment by Soywei00** [comment]
![image](https://user-images.githubusercontent.com/16738821/155159577-96d17e21-47db-490c-8584-bde0bff391ce.png)
服务-定制导出里面没有外部链接和内部链接的选项。
简悦2.2.0，同步助手1.0.1

**Reply by Kenshin** [reply]
开启这个选项。

![image](https://user-images.githubusercontent.com/81074/155253470-a6045f81-1179-47b5-b0cb-954946365863.png)
